### DISCUSSION TITLE
Use the ntp to show time locally on screen (solved)

### DISCUSSION THREAD

**Discussion by hb020** [HASwitchPlate/openHASP#595] [Ideas]
(disclaimer: am new to openHASP, found no way to do what I want to do, but it may be possible already, I just haven't found it)

Now that openHASP has ntp, why not allow the showing of date/time locally, without relying on MQTT?

EDIT: this is already solved via template, see comment below.

Problems to be solved:
1. new "magic strings" (like the strings %hostname%, %ip%, %mac%,.... )
2. the formatting of those date/time strings. Proposition:
 * either a series of magic strings, like "%date_MMDD%, %date_DDMM%, %time_hhmm% (24h format), and something else for those who want AM/PM
 * either separate run time formatter support via attributes. EDIT: it is "template"
 * either compilation time options for the formatting
3. force regular refresh without it being pushed from MQTT or other commands. This refresh functionality may also be very useful for img.src=http://....., where the remote image changes regularly. (cameras)

**Comment by smcgann99** [comment]
This is already possible - supported formats https://cplusplus.com/reference/ctime/strftime/

An example -
`{ "page": 2, "id": 251, "obj": "label", "x": 165, "y": -13, "w": 150, "h": 57, "text_font": 58, "text_color": "#969696", "align": "center", "text": "98:80:00", "template": "%H:%M" },

{ "page": 2, "id": 252, "obj": "label", "x": 165, "y": 45, "w": 150,"bg_color":"blue","bg_opa":100, "h": 32, "text_font": 28, "text_color": "#969696", "align": "center", "text": "Wednesday", "template": "%a %d %b" }`

**Reply by hb020** [reply]
That's great, thank you, didn't know it. But this only the formatting. How about the 2 other points? (edited the original post to reflect your addition)

**Reply by smcgann99** [reply]
That formatting is the "magic" string. It doesn't need to be refreshed it updates itself, even by the second. Just try the example...

**Reply by hb020** [reply]
WOW, that is great. Thanks. Is it something that I overlooked in the documentation, or is it not (yet) there?

**Reply by smcgann99** [reply]
Undocumented as yet.

**Reply by hb020** [reply]
"template" might be misleading, as one day one might want to format incoming integers or floats also. it might be better to rename that to "dt_template" or "strftime" or to move that inside the text field in some form.
